server.py

```diff
@@ -292,80 +303,82 @@ def get_lang_match_score(c: Coach, languages: Dict[str, int]) -> int:
 
 
 @app.route("/")
-async def index(request: sanic.request.Request) -> sanic.response.HTTPResponse:
-    return await sanic.response.file("build/index.html")
+def index() -> flask.Response:
+    return app.send_static_file("index.html")
 
 
-@app.route("/api/v1/coaches", methods=["GET", "POST", "OPTIONS"])
-@app.route("/api/v1/coaches/<coach_id:int>", methods=["GET", "POST", "DELETE"])
-async def api_coaches(
-    request: sanic.request.Request, coach_id: Optional[int] = None
-) -> sanic.response.HTTPResponse:
+@app.route("/api/v1/coaches", methods=["GET", "POST"])
+def api_coaches() -> FlaskReturn:
     """HTTP API for 'coaches'."""
-    headers = COMMON_HEADERS
+    headers = COMMON_HEADERS.copy()
+
+    try:
+        if flask.request.method == "GET":
+            coaches = get_coaches()
+            response = flask.jsonify([c.to_json() for c in coaches])
+        elif flask.request.method == "POST":
+            coach = create_coach(flask.request.json)
+            response = flask.jsonify(coach.to_json())
+        else:
+            assert False, "Unsupported method"
+    except Exception:
+        logger.exception("Unexpected exception on %s route", flask.request.path)
```

Comment:
At the moment, this try/except is very general - I think we should also at least log the traceback/exception type here.

Reply:
(Same comment for all routes!)

Reply:
Logger.exception() logs the full traceback - the exc_info=True we had before was doing nothing. FWIW I think a better solution would be to add a general exception handler to avoid duplicating this logic (possible with flask I think), but that's out of scope.